nuxeo-core/nuxeo-core-storage-sql/nuxeo-core-storage-sql/src/main/java/org/nuxeo/ecm/core/storage/sql/jdbc/JDBCRowMapper.java

```diff
@@ -1396,7 +1396,24 @@ protected Boolean copyRows(String tableName, Set<Serializable> ids, Map<Serializ
         List<Serializable> todo = new ArrayList<>(Collections.singleton(rootId));
         List<NodeInfo> descendants = new ArrayList<NodeInfo>();
         while (!todo.isEmpty()) {
-            List<NodeInfo> infos = getChildrenNodeInfos(todo);
+            List<NodeInfo> infos;
+            int size = todo.size();
+            int chunkSize = sqlInfo.getMaximumArgsForIn();
+            if (size > chunkSize) {
```

Comment:
may be this chunk logic could have been moved to sub method in getChildrenNodeInfos, otherwise +1